File: Product of Array Except Self - 238/ans.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int product = 1;
        int zero = 0;
        for (int i = 0; i < nums.size(); ++i){
            if (nums[i] == 0)
                zero++;
            else
                product *= nums[i];
        }

        vector<int> ans(nums.size());
        
        for (int i = 0; i < nums.size(); ++i){
            if (zero > 1){
                ans[i] = 0;
            }else if (zero == 1){
                if (nums[i] == 0)
                    ans[i] = product;
                else
                    ans[i] = 0;
            }else{
                ans[i] = product / nums[i];
            }
        }
        return ans;
    }
};

File: Remove Duplicates from Sorted Array II - 80/ans.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0)
            return 0;
        int c = 1;
        int location = 1;
        for (int i = 1; i < nums.length; ++i){
            if (nums[i] == nums[i - 1])
                c++;
            else
                c = 1;
            if (c <= 2){
                nums[location++] = nums[i];
            }
        }
        return location;
    }
}
